backends/advanced/src/advanced_omi_backend/utils/file_utils.py

```
@@ -0,0 +1,82 @@
+        raise FileNotFoundError(error_msg)
+
+    # Create extraction directory if needed
+    if create_extract_dir:
+        extract_dir.mkdir(parents=True, exist_ok=True)
+
+    # Extract zip file
+    try:
+        with zipfile.ZipFile(zip_path, 'r') as zf:
+            zf.extractall(extract_dir)
+        logger.info(f"Successfully extracted {zip_path} to {extract_dir}")
+        return extract_dir
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)